File: Algorithm/Other/at coder/Other/A36.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int N = sc.nextInt();
        int K = sc.nextInt();
        
        int minSteps = 2 * (N - 1);
        
        if (K >= minSteps && (K - minSteps) % 2 == 0) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
        
        sc.close();
    }
}

File: Algorithm/Other/at coder/Other/A36/A36.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	var N, K int

	// 標準入力を読み込む
	scanner := bufio.NewScanner(os.Stdin)
	if scanner.Scan() {
		fmt.Sscanf(scanner.Text(), "%d %d", &N, &K)
	}

	minSteps := 2 * (N - 1)

	if K >= minSteps && (K-minSteps)%2 == 0 {
		fmt.Println("Yes")
	} else {
		fmt.Println("No")
	}
}
